test-case/verify-tplg-binary.sh

```diff
@@ -19,19 +19,18 @@ set -e
 # shellcheck source=case-lib/lib.sh
 source "$(dirname "${BASH_SOURCE[0]}")"/../case-lib/lib.sh
 
-OPT_NAME['t']='tplg'     OPT_DESC['t']='tplg file, default value is env TPLG: $''TPLG'
+OPT_NAME['t']='tplg'     OPT_DESC['t']="tplg file, default value is env TPLG: $TPLG"
 OPT_HAS_ARG['t']=1         OPT_VAL['t']="$TPLG"
 
 func_opt_parse_option "$@"
 setup_kernel_check_point
 tplg=${OPT_VAL['t']}
 
-tplg_path=`func_lib_get_tplg_path "$tplg"`
-[[ "$?" != "0" ]] && die "No available topology for this test case"
+tplg_path=$(func_lib_get_tplg_path "$tplg") || die "No available topology ($tplg) for this test case"
 
 dlogi "Checking topology file: $tplg_path"
-dlogi "Found file: $(md5sum $tplg_path|awk '{print $2, $1;}')"
-tplgData=$(sof-tplgreader.py $tplg_path 2>/dev/null)
+dlogi "Found file: $(md5sum "$tplg_path"|awk '{print $2, $1;}')"
+tplgData=$(sof-tplgreader.py "$tplg_path" 2>/dev/null)
```

Comment:
- NEVER discard stderr.
- Unlike shell, Python is a safe language and we can trust its exit codes. Checking for empty strings is really ugly.
```suggestion
tplgData=$(sof-tplgreader.py "$tplg_path") ||
   die "No valid pipeline(s) found in $tplg_path"
```

Reply:
So we're still leaving the following line untouched?
`[[ -z "$tplgData" ]] && die "No valid pipeline(s) found in $tplg_path"`

Got it.